File: app/src/main/java/com/dumuzeyn/mp3player/PlaybackStateProvider.java
package com.dumuzeyn.mp3player;

import androidx.annotation.Nullable;
import java.util.List;

/** Read-only Media3 projection used by player UI. */
interface PlaybackStateProvider {
    PlaybackSnapshot currentSnapshot();

    @Nullable
    Track currentTrack();

    boolean isCurrentTrack(Track track);

    boolean isPlaying();

    int repeatMode();

    List<Track> activeQueue();

    int queueIndex(Track track);
}

File: app/src/main/java/com/dumuzeyn/mp3player/PlaybackStateProvider.kt
package com.dumuzeyn.mp3player

import kotlinx.coroutines.flow.StateFlow

/** Read-only Media3 projection used by player UI. */
interface PlaybackStateProvider {
    fun state(): StateFlow<PlaybackSnapshot>
    fun currentSnapshot(): PlaybackSnapshot
    fun currentTrack(): Track?
    fun isCurrentTrack(track: Track?): Boolean
    fun isPlaying(): Boolean
    fun repeatMode(): Int
    fun activeQueue(): List<Track>
    fun queueIndex(track: Track): Int
}
